### PR TITLE
add the process and thread scopes

### PR DESCRIPTION
**Pull request by yelhamer**
this PR is for adding the process scope as discussed here: #1517

there isn't any separation between static-flavor only subscopes and dynamic-flavor only ones, and I think that it should be implemented when the flavor keyword gets added. I'd love to hear your opinions on this however.

### Checklist

- [ ] No CHANGELOG update needed

- [ ] No new tests needed

- [ ] No documentation update needed

### CODE REVIEW

**Review by mr-tz — COMMENTED** [review]
looks good, overall, I've added a few comments and questions

File: tests/test_main.py [hunk 1]
```diff
@@ -133,11 +133,24 @@ def test_ruleset():
                     """
                 )
             ),
+            capa.rules.Rule.from_yaml(
+                textwrap.dedent(
+                    """
+                    rule:
+                        meta:
+                            name: process rule
+                            scope: process
```
Comment (williballenthin) [inline tests/test_main.py:142]: process should not be valid as a static-flavor scope. i see that you haven't split out the static/dynamic flavors yet, so i realize you can't test this quite yet.

File: capa/rules/__init__.py [hunk 2]
```diff
@@ -106,6 +108,12 @@ class Scope(str, Enum):
         capa.features.common.Namespace,
         capa.features.common.Characteristic("mixed mode"),
     },
+    PROCESS_SCOPE: {
+        capa.features.common.String,
```
[the remark this answers is not in the document]
Comment (yelhamer) [reply capa/rules/__init__.py:112]: regarding the thread scope features, should I add them in this PR and rename it?

[the remark this answers is not in the document]
Comment (williballenthin) [reply capa/rules/__init__.py:112]: yeah lets go ahead and add them here, too

File: capa/rules/__init__.py [hunk 3]
```diff
@@ -150,7 +169,11 @@ class Scope(str, Enum):
 SUPPORTED_FEATURES[BASIC_BLOCK_SCOPE].update(SUPPORTED_FEATURES[GLOBAL_SCOPE])
 SUPPORTED_FEATURES[FUNCTION_SCOPE].update(SUPPORTED_FEATURES[GLOBAL_SCOPE])
 SUPPORTED_FEATURES[FILE_SCOPE].update(SUPPORTED_FEATURES[GLOBAL_SCOPE])
+SUPPORTED_FEATURES[PROCESS_SCOPE].update(SUPPORTED_FEATURES[GLOBAL_SCOPE])
+SUPPORTED_FEATURES[THREAD_SCOPE].update(SUPPORTED_FEATURES[GLOBAL_SCOPE])
 
+# all thread scope features are also function features
+SUPPORTED_FEATURES[FUNCTION_SCOPE].update(SUPPORTED_FEATURES[THREAD_SCOPE])
```
[the remark this answers is not in the document]
Comment (yelhamer) [reply capa/rules/__init__.py]: ah sorry, that was a mistake. I've corrected now.

**Pull request merged**